crates/bevy_ecs/src/world/mod.rs

```diff
@@ -1333,6 +1339,37 @@ impl Default for MainThreadValidator {
     }
 }
 
+pub trait ManyRes<'w> {
+    type Item;
+
+    fn get_resources(world: &'w World) -> Self::Item;
+}
+
+pub trait ManyResMut<'w> {
+    type Item;
+    fn get_resources_mut(world: &'w mut World) -> Self::Item;
+}
+
+macro_rules! impl_tuple_res {
+    ($($name: ident),*) => {
+        impl<'w, $($name: Resource,)*> ManyRes<'w> for ($($name,)*){
+            type Item = ($(Option<&'w $name>,)*);
+            #[allow(unused_variables, unused_unsafe, clippy::unused_unit)]
+            fn get_resources(world: &'w World) -> Self::Item {
+                ($(world.get_resource::<$name>(),)*)
+            }
+        }
+        impl<'w, $($name: Resource,)*> ManyResMut<'w> for ($($name,)*){
+            type Item = ($(Option<Mut<'w, $name>>,)*);
+            #[allow(unused_variables, unused_unsafe, clippy::unused_unit)]
+            fn get_resources_mut(world: &'w mut World) -> Self::Item {
+                unsafe{ ($(world.get_resource_unchecked_mut::<$name>(),)*) }
```

Comment:
The suggested safety comment is currently a lie :p We can get UB here by requesting two resources of the same type.

You'll want to check them for uniqueness instead.

There are three options here:
1. Compare `TypeId`. This is very simple, but may not be future-proof if we add "dynamic resources".
2. Compare `ComponentId`. Probably more correct.
3. Check that their accesses are compatible. Probably overkill, given how simple resources are.

This PR should also include a test that this flavor of UB is impossible.

Reply:
I'm not quite sure how to achieve the comparisons in the context of a macro. 🤔